src/bullet.py

```diff
@@ -1,48 +1,87 @@
 from __future__ import annotations
 
-import math
-from typing import TYPE_CHECKING, List, cast
+from typing import List, Tuple, cast
 
 import arcade
-
 from arena.wall import Wall
 from constants import SCREEN_HEIGHT, SCREEN_WIDTH
-from iron_math import add_vec, move_sprite_relative_to_parent, rotate_vec
+from iron_math import move_sprite_polar, set_sprite_location
 from linked_sprite import LinkedSprite
 from sprite_lists import SpriteLists
 
-# This allows a circular import only for the purposes of type hints
-# Weapon will never create and instance of Player
-if TYPE_CHECKING:
-    from weapon import Weapon
-
 
 class Projectile:
-    def __init__(self, sprite: LinkedSprite[Projectile], sprite_lists: SpriteLists):
-        self.sprite = sprite
-        sprite.owner = self
-        sprite_lists.projectiles.append(self.sprite)
-        self.damage = 0
+    """
+    A projectile can be anything that is created by a weapon or another projectile included beams, rockets, etc.
+    It stores the characteristics of the projectile and is responible the associated sprite
+    """
 
+    sprite: LinkedSprite[Projectile]
+    sprite_lists: SpriteLists
+    damage: float
+    speed: float
+    radians: float
+    sprite_rotation_offset: float
+    exists: bool
+    beam: bool
+    beam_range: float
+    explodes: bool
 
-class Beam:
     def __init__(
-        self, sprite: LinkedSprite[Beam], sprite_lists: SpriteLists, weapon: Weapon
+        self,
+        sprite: LinkedSprite[Projectile],
+        sprite_lists: SpriteLists,
+        speed: float,
+        radians: float,
+        damage: float,
     ):
         self.sprite = sprite
         sprite.owner = self
-        sprite_lists.beams.append(self.sprite)
-        self.dps = 0
-        self.weapon = weapon
+        self.sprite_lists = sprite_lists
+        self.speed = speed
+        self.radians = radians
+        self.damage = damage
+        self.beam = False
+        self.beam_range = 0
+        self.explodes = False
+        self.sprite_rotation_offset = 0
+        self.append_sprite()
+
+    def update(self, delta_time: float):
+        move_sprite_polar(self.sprite, self.speed * delta_time, self.radians)
+
+    def set_location(self, location: Tuple[float, float, float]):
+        set_sprite_location(self.sprite, location)
+        self.sprite.radians += self.sprite_rotation_offset
 
+    def set_beam(self, range: float):
+        self.beam = True
+        self.beam_range = range
 
-# For naming purposes, a bullet can be anything that comes out of a weapon included beams, rockets, etc
-def bullet_behavior(
+    def set_sprite_rotation_offset(self, radians: float):
+        self.sprite_rotation_offset = radians
+
+    def set_explodes(self):
+        self.explodes = True
```

Comment:
Remind me again, where is self.explodes being used?  Is that for future stuff?

I also think we can eliminate this method.  And if we need to do logic in the future, use `@property`

Reply:
Definitively groundwork for future stuff that can be implemented after the basic collision code is in place